### PR TITLE
[FLINK-3341] Make 'auto.offset.reset' compatible with Kafka 0.8 and 0.9

### PR DESCRIPTION
**Pull request by rmetzger**

### CODE REVIEW

**Comment by StephanEwen** [comment]
Good fix, pretty important to get this into 1.0

File: flink-streaming-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/LegacyFetcher.java [hunk 1]
```diff
@@ -576,7 +576,8 @@ private static void getLastOffset(SimpleConsumer consumer, List<FetchPartition>
 
 		private static long getInvalidOffsetBehavior(Properties config) {
 			long timeType;
-			if (config.getProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "latest").equals("latest")) {
+			String val = config.getProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "largest");
+			if (val.equals("largest") || val.equals("latest")) { // largest is kafka 0.8, latest is kafka 0.9
```
Comment (shikhar) [inline flink-streaming-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/LegacyFetcher.java]: > latest is kafka 0.9

I think 'latest' is just flink :)

Comment (rmetzger) [inline flink-streaming-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/LegacyFetcher.java]: No. Check the new consumer config of Kafka 0.9:
http://kafka.apache.org/documentation.html#newconsumerconfigs

**Comment by StephanEwen** [comment]
Merging this...

**Pull request closed without merging**